pkg/operator/ceph/cluster/controller.go

```diff
@@ -981,3 +969,56 @@ func validateExternalClusterSpec(cluster *cluster) error {
 
 	return nil
 }
+
+// ValidateCephVersionsBetweenLocalAndExternalClusters validates
+func ValidateCephVersionsBetweenLocalAndExternalClusters(localVersion, externalVersion cephver.CephVersion) error {
+	// We only support Nautilus or newer
+	if !externalVersion.IsAtLeastNautilus() {
+		return fmt.Errorf("unsupported ceph version %s, need at least nautilus, delete your cluster CR and create a new one with a correct ceph version", externalVersion.String())
+	}
+
+	// Identical version, regardless if other CRs are running, it's ok!
+	if cephver.IsIdentical(localVersion, externalVersion) {
+		return nil
+	}
+
+	// Local version must never be higher than the external one
+	if cephver.IsSuperior(localVersion, externalVersion) {
+		return fmt.Errorf("local cluster ceph version is higher %s than the external cluster %s, this must never happen", externalVersion.String(), localVersion.String())
+	}
+
+	// External cluster was updated to a minor version higher, consider updating too!
+	if localVersion.Major == externalVersion.Major {
+		if cephver.IsSuperior(externalVersion, localVersion) {
+			logger.Warningf("external cluster ceph version is a minor version higher %s than the local cluster %s, consider upgrading", externalVersion.String(), localVersion.String())
+			return nil
+		}
+	}
+
+	// The external cluster was upgraded, consider upgrading too!
+	if localVersion.Major < externalVersion.Major {
+		return fmt.Errorf("external cluster ceph version is a major version higher %s than the local cluster %s, consider upgrading", externalVersion.String(), localVersion.String())
+	}
+
+	logger.Debugf("local version is %s, external version is %s", localVersion.String(), externalVersion.String())
+	return nil
+}
+
+// Add validation in the code to fail if the external cluster has no OSDs keep waiting
+func populateExternalClusterInfo(context *clusterd.Context, namespace string) *cephconfig.ClusterInfo {
+	var clusterInfo *cephconfig.ClusterInfo
+	for {
+		var err error
+		clusterInfo, _, _, err = mon.LoadClusterInfo(context, namespace)
+		if err != nil {
+			logger.Warningf("waiting for the connection info of the external cluster. %+v", err)
+			time.Sleep(10 * time.Second)
```

Comment:
Should we make this interval longer? Or else suppress the logging so we don't continue to log the same message with every retry.

Reply:
Every minute?

Reply:
I believe the logging is useful here.

Reply:
What if we only log occasionally (every 10th retry) instead of every time and still retry every 10s?

Reply:
Pushed to 1m and kept the logging.